crates/ra_analysis/tests/tests.rs

```diff
@@ -2,14 +2,30 @@ use ra_syntax::TextRange;
 use test_utils::{assert_eq_dbg, assert_eq_text};
 
 use ra_analysis::{
-    mock_analysis::{analysis_and_position, single_file, single_file_with_position, MockAnalysis},
+    mock_analysis::{analysis_and_position, analysis_and_range, single_file, single_file_with_position, MockAnalysis},
     AnalysisChange, CrateGraph, FileId, FnSignatureInfo,
 };
 
 fn get_signature(text: &str) -> (FnSignatureInfo, Option<usize>) {
     let (analysis, position) = single_file_with_position(text);
     analysis.resolve_callable(position).unwrap().unwrap()
 }
+#[test]
+fn test_type_of() {
+    let (analysis, range) = analysis_and_range(
+        "
+        //- /lib.rs
+        pub fn foo() -> u32 {
+            1
+        };
+
+        let <|>foo_test<|> = foo();
+    ",
```

Comment:
There are two reasons why this doesn't work.

* this is just an invalid syntax: the `let` is on the top-level, and one can't have a top-level `let` in Rust.
* `type_of` gets the type of the *expression* at the moment, it doesn't work for bindings. 

So, the following does work:

```
let (analysis, range) = single_file_with_range(
        "
        pub fn foo() -> u32 { 1 };

        fn main() {
            let foo_test = <|>foo()<|>;
        }
        ",
    );
```